data/generate_embeddings.py

```diff
@@ -0,0 +1,598 @@
+        batch = chunks[i : i + batch_size]
+        
+        try:
+            vectors = provider.embed_batch(
+                batch,
+                retry_attempts=retry_attempts,
+                retry_min_wait=retry_min_wait,
+                retry_max_wait=retry_max_wait
+            )
+            
+            # Validate embeddings
+            if not validate_embeddings(vectors, len(batch)):
+                print(f"  Error: Validation failed for batch starting at chunk {i}")
+                continue
+            
+            # Save chunks
+            for j, (text, vec) in enumerate(zip(batch, vectors)):
+                save_chunk_json(
+                    output_dir=output_dir,
+                    city=city,
+                    chunk_id=start_index + total_written + j,
+                    text=text,
+                    embedding=vec,
+                    source_file=input_path.name,
+                    model=provider.model,
+                )
+            
+            total_written += len(batch)
+            print(f"  Processed batch {i//batch_size + 1}: {len(batch)} chunks")
+            
+            # Polite pacing
+            time.sleep(polite_delay)
+            
+        except Exception as e:
+            print(f"  Error processing batch {i//batch_size + 1}: {e}")
+            continue
+
+    last_index = start_index + total_written - 1 if total_written > 0 else None
+    print(f"  Completed: {total_written} chunks written for city '{city}'")
+    
+    return total_written, last_index
+
+# -------------------------
+# CLI & main
+# -------------------------
+def parse_args() -> argparse.Namespace:
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(
+        description="Generate semantic embeddings for travel content."
+    )
+    parser.add_argument(
+        "--input-dir", 
+        type=str, 
+        default=str(Path(DEFAULT_INPUT_DIR)), 
+        help="Directory with source documents"
+    )
+    parser.add_argument(
+        "--output-dir", 
+        type=str, 
+        default=str(Path(DEFAULT_OUTPUT_DIR)), 
+        help="Directory to write chunk JSON files"
+    )
+    parser.add_argument(
+        "--provider", 
+        type=str, 
+        default="openai", 
+        choices=["openai"], 
+        help="Embedding provider"
+    )
+    parser.add_argument(
+        "--model", 
+        type=str, 
+        default=DEFAULT_MODEL, 
+        help="Embedding model name"
+    )
+    parser.add_argument(
+        "--max-tokens", 
+        type=int, 
+        default=DEFAULT_MAX_TOKENS, 
+        help="Max tokens per chunk"
+    )
+    parser.add_argument(
+        "--overlap", 
+        type=float, 
+        default=DEFAULT_OVERLAP, 
+        help="Chunk overlap ratio (0-1)"
+    )
+    parser.add_argument(
+        "--batch-size", 
+        type=int, 
+        default=DEFAULT_BATCH_SIZE, 
+        help="Batch size for embedding API calls"
+    )
+    return parser.parse_args()
```

Comment:
Add argument for controlling retry parameters and polite delay for flexibility.
Document CLI usage with examples.